doc/build/dts/bindings-intro.rst

```diff
@@ -0,0 +1,134 @@
+        num-foos = <3>;
+   };
+
+Here is a minimal binding file which matches the node:
+
+.. code-block:: yaml
+
+   # A YAML binding matching the node
+
+   compatible: "foo-company,bar-device"
+
+   properties:
+     num-foos:
+       type: int
+       required: true
+
+The build system matches the ``bar-device`` node to its YAML binding because
+the node's ``compatible`` property matches the binding's ``compatible:`` line.
+
+What the build system does with bindings
+****************************************
+
+The build system uses bindings both to validate devicetree nodes and to convert
+the devicetree's contents into the generated :ref:`devicetree_generated.h
+<dt-outputs>` header file.
+
+For example, the build system would use the above binding to check that the
+required ``num-foos`` property is present in the ``bar-device`` node, and that
+its value, ``<3>``, has the correct type.
+
+The build system will then generate a macro for the ``bar-device`` node's
+``num-foos`` property, which will expand to the integer literal ``3``. This
+macro lets you get the value of the property in C code using the API which is
+discussed later in this guide in :ref:`dt-from-c`.
+
+For another example, the following node would cause a build error, because it
+has no ``num-foos`` property, and this property is marked required in the
+binding:
+
+.. code-block:: devicetree
+
+   bad-node {
+   	compatible = "foo-company,bar-device";
+   };
+
+Other ways nodes are matched to bindings
+****************************************
+
+If a node has more than one string in its ``compatible`` property, the build
+system looks for compatible bindings in the listed order and uses the first
+match.
+
+Take this node as an example:
+
+.. code-block:: devicetree
+
+   baz-device {
+   	compatible = "foo-company,baz-device", "generic-baz-device";
+   };
+
+The ``baz-device`` node would get matched to a binding with a ``compatible:
+"generic-baz-device"`` line if the build system can't find a binding with a
+``compatible: "foo-company,baz-device"`` line.
+
+Nodes without compatible properties can be matched to bindings associated with
+their parent nodes. These are called "child bindings". If a node describes
+hardware on a bus, like I2C or SPI, then the bus type is also taken into
+account when matching nodes to bindings. (See :ref:`dt-bindings-on-bus` for
+details).
+
+See :ref:`dt-zephyr-user` for information about a special node that doesn't
+require any binding.
+
+.. _dt-where-bindings-are-located:
+
+Where bindings are located
+**************************
+
+Binding file names usually match their ``compatible:`` lines. For example, the
+above example binding would be named :file:`foo-company,bar-device.yaml` by
+convention.
+
+The build system looks for bindings in :file:`dts/bindings`
+subdirectories of the following places:
+
+- the zephyr repository
+- your :ref:`application source directory <application>`
+- your :ref:`board directory <board_porting_guide>`
```

Comment:
Can you cite your source?

Reply:
cmake/modules/dts.cmake line 51:

```
# 'DTS_ROOT' is a list of directories where a directory tree with DT
# files may be found. It always includes the application directory,
# the board directory, and ${ZEPHYR_BASE}.
list(APPEND
  DTS_ROOT
  ${APPLICATION_SOURCE_DIR}
  ${BOARD_DIR}
  ${SHIELD_DIRS}
  ${ZEPHYR_BASE}
  )
```

Reply:
Interesting that this is done but unused upstream (or in NCS). Added now as a separate commit since this gap in the docs existed before this PR (you're seeing it in green because this content was moved from another file). Thanks!